replication-api-impl/src/main/java/com/connexta/replication/api/impl/worker/Worker.java

```diff
@@ -0,0 +1,259 @@
+    this.running = false;
+    try {
+      Thread.currentThread().join();
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+    }
+  }
+
+  /**
+   * @return {@code true} if this worker is transferring an item, otherwise {@code false}
+   */
+  public boolean isProcessing() {
+    return processing.get();
+  }
+
+  private boolean supported(Task task) throws InterruptedException {
+    if (!OperationType.UNKNOWN.equals(task.getOperation())) {
+      task.fail(ErrorCode.UNSUPPORTED_ERROR,
+          String.format("Operation task %s is not supported by this worker", task));
+      return false;
+    }
+    return true;
+  }
+
+  private void handle(Task task) throws InterruptedException {
+    boolean proceed = checkAdapterAvailabilities(task);
+    if (!proceed) {
+      return;
+    }
+
+    List<DDFMetadataInfo> infos = task.metadatas()
+        .filter(t -> t instanceof DDFMetadataInfo)
+        .map(DDFMetadataInfo.class::cast)
+        .collect(Collectors.toList());
+
+    if (infos.size() > 1) {
+      task.fail(ErrorCode.UNSUPPORTED_ERROR,
+          "more than one metadata is currently not supported by this worker");
+      return;
+    }
+
+    if(infos.isEmpty()) {
+      task.fail(ErrorCode.UNSUPPORTED_ERROR,
+          "unable to handle any of the metadatas received in this task");
+      return;
+    }
+
+    final DDFMetadataInfo info = infos.get(0);
+    final Metadata metadata = new MetadataImpl(((Map) info.getData()).get("raw-metadata"),
+        info.getDataClass(), task.getId(),
+        Date.from(info.getLastModified()));
+
+    final OperationType type = task.getOperation();
+    switch (type) {
+      case HARVEST:
+      case REPLICATE_FROM:
+        doCreate(task, metadata, destination, local);
+        break;
+      case REPLICATE_TO:
+        doCreate(task, metadata, local, destination);
+        break;
+      case DELETE:
+        // always delete on non-local site
+        doDelete(task, metadata);
+        break;
+      default:
+        // should never hit this since deserialization will return UNKNOWN
+        task.fail(ErrorCode.UNSUPPORTED_ERROR,
+            String.format("Operation task %s is not supported by this worker", task));
+    }
+  }
+
+  private void doCreate(final Task task, final Metadata metadata, final NodeAdapter source,
+      final NodeAdapter destination)
+      throws InterruptedException {
+    final String metadataId = task.getId();
+    final String operation = task.getOperation().name();
+    if (source.exists(metadata)) {
+      CreateRequest request = new CreateRequestImpl(List.of(metadata));
+      boolean success = destination.createRequest(request);
+      if (!success) {
+        task.fail(ErrorCode.OPERATION_FAILURE,
+            String.format("Failed to %s metadata %s from %s to %s",
+                operation,
+                metadataId,
+                source.getSystemName(),
+                destination.getSystemName()));
```

Comment:
❗️ Looks like it might fall-through below where the task will end up being marked completed. If it failed then that is it and we should return